testing/src/scenario/_runtime.py

```diff
@@ -352,10 +352,14 @@ def exec(
                 ) from e
 
             finally:
-                if previous_juju_version is None:
-                    del os.environ["JUJU_VERSION"]
-                else:
-                    os.environ["JUJU_VERSION"] = previous_juju_version
+                for key, value in os.environ.copy().items():
+                    if key in previous_env:
+                        os.environ[key] = value
+                    else:
+                        del os.environ[key]
+                for key, value in previous_env.items():
+                    if key not in os.environ:
+                        os.environ[key] = value
```

Comment:
```suggestion
                for key, value in previous_env.items():
                        os.environ[key] = value
                for key in tuple(os.environ):
                    if key not in previous_env:
                        del os.environ[key]
```

Reply:
Oh hold up, how about this:
```suggestion
                for key in tuple(os.environ):
                    if key not in previous_env:
                        del os.environ[key]
                os.environ.update(previous_env)
```

Reply:
Ah, yes, that's much nicer, thanks!